snap/local/hooks/cmd/configure/configure.go

```diff
@@ -127,113 +342,200 @@ func handleAllServices() error {
-				return err
-			}
-			if err = handleSingleService("kong-daemon", status); err != nil {
-				return err
-			}
-			if err = handleSingleService("security-proxy-setup", status); err != nil {
-				return err
-			}
-
-			// additionally, the security-proxy needs to use the following
-			// services:
-			// - vault (because security-proxy-setup will access/store secrets in vault)
-			// - security-secretstore-setup
-			// so if we are turning the security-api-gateway on, then turn
-			// those services on too
-			if status == ON {
-				if err = handleSingleService("vault", ON); err != nil {
-					return err
-				}
-				if err = handleSingleService("security-secretstore-setup", ON); err != nil {
+			switch status {
+			case ON:
+				// NOTE: the original bash based implementation would
+				// additionally handle the secret-store dependency.
+				// Due to the added complexity, this initial implementation
+				// does not automatically handle enabling the secret-store
+				// if/when the proxy is dynamically enabled.
+				if !secretStoreActive {
+					err = fmt.Errorf("edgexfoundry:configure security-proxy=on not allowed;" +
+						"secret-store=off")
 					return err
 				}
+
+				fallthrough
+			case OFF:
+				serviceList = getProxyServices()
+			case UNSET:
+				// this is the default status of all services if no
+				// configuration has been specified; no-op
+				continue
+			default:
+				return fmt.Errorf("edgexfoundry:configure: invalid value for security-proxy: %s", status)
 			}
+
 		case secStoreService:
 			hooks.Debug("edgexfoundry:configure: secretstore")
-			// the security-api-gateway consists of the following services:
-			// - vault
-			// - security-secretstore-setup
-			// and since the security-api-gateway needs to be able to use
-			// security-secret-store, we also need to turn off those services
-			// if this one is disabled
-			if status == OFF {
-				if err = handleSingleService("postgres", OFF); err != nil {
-					return err
-				}
-				if err = handleSingleService("kong-daemon", OFF); err != nil {
-					return err
-				}
-				if err = handleSingleService("security-proxy-setup", OFF); err != nil {
+
+			switch status {
+			case ON:
+				return fmt.Errorf("edgexfoundry:configure security-secret-store=on not allowed")
+			case OFF:
+				secretStoreActive = false
+
+				if err = disableSecretStoreAndRestart(); err != nil {
 					return err
 				}
-
-			}
-			if err = handleSingleService("vault", status); err != nil {
-				return err
-			}
-			if err = handleSingleService("security-secretstore-setup", status); err != nil {
-				return err
+				continue
+			case UNSET:
+				// this is the default status of all services if no
+				// configuration has been specified; no-op
+				continue
+			default:
+				return fmt.Errorf("edgexfoundry:configure: invalid value for security-secret-store: %s", status)
 			}
+
 		default:
 			hooks.Debug("edgexfoundry:configure: other service")
```

Comment:
Not from this PR. But I'd add here the name of that other service:
`hooks.Debug("edgexfoundry:configure: other service: "+s)`